src/posix/platform/netif.cpp

```diff
@@ -596,18 +1319,179 @@ void platformNetifInit(otInstance *aInstance, const char *aInterfaceName)
+                                       char *      deviceName,
+                                       size_t      deviceNameLen)
+{
+    int         flags = IFF_BROADCAST | IFF_MULTICAST;
+    int         err;
+    const char *last_slash;
+    const char *path;
+
+    (void)aInterfaceName;
+    (void)aInstance;
+
+    path = OPENTHREAD_POSIX_TUN_DEVICE;
+
+    sTunFd = open(path, O_RDWR | O_NONBLOCK);
+    VerifyOrDie(sTunFd >= 0, OT_EXIT_ERROR_ERRNO);
+
+#if defined(__NetBSD__) || defined(__FreeBSD__)
+    err = ioctl(sTunFd, TUNSIFMODE, &flags);
+    VerifyOrDie(err == 0, OT_EXIT_ERROR_ERRNO);
+#endif
+
+    flags = 1;
+    err   = ioctl(sTunFd, TUNSIFHEAD, &flags);
+    VerifyOrDie(err == 0, OT_EXIT_ERROR_ERRNO);
+
+    last_slash = strrchr(OPENTHREAD_POSIX_TUN_DEVICE, '/');
+    VerifyOrDie(last_slash != NULL, OT_EXIT_ERROR_ERRNO);
+    last_slash++;
+
+    strncpy(deviceName, last_slash, deviceNameLen);
+}
+#endif
+
+static void platformConfigureNetLink(void)
+{
+#if defined(__linux__)
+    sNetlinkFd = socket(AF_NETLINK, SOCK_DGRAM, NETLINK_ROUTE);
+#elif defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
+    sNetlinkFd = socket(PF_ROUTE, SOCK_RAW, 0);
+#else
+#error "!! Unknown platform !!"
+#endif
+    VerifyOrDie(sNetlinkFd >= 0, OT_EXIT_ERROR_ERRNO);
+
+#if defined(__linux__)
+    {
+        struct sockaddr_nl sa;
+
+        memset(&sa, 0, sizeof(sa));
+        sa.nl_family = AF_NETLINK;
+        sa.nl_groups = RTMGRP_LINK | RTMGRP_IPV6_IFADDR;
+        VerifyOrDie(bind(sNetlinkFd, reinterpret_cast<struct sockaddr *>(&sa), sizeof(sa)) == 0, OT_EXIT_ERROR_ERRNO);
+    }
+#endif
+
+#if defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
+    {
+        int status;
+#ifdef ROUTE_FILTER
+        unsigned int msgfilter = ROUTE_FILTER(RTM_IFINFO) | ROUTE_FILTER(RTM_NEWADDR) | ROUTE_FILTER(RTM_DELADDR) |
+                                 ROUTE_FILTER(RTM_NEWMADDR) | ROUTE_FILTER(RTM_DELMADDR);
+#define FILTER_CMD ROUTE_MSGFILTER
+#define FILTER_ARG msgfilter
+#define FILTER_ARG_SZ sizeof(msgfilter)
+#endif
+#ifdef RO_MSGFILTER
+        uint8_t msgfilter[] = {RTM_IFINFO, RTM_NEWADDR, RTM_DELADDR};
+#define FILTER_CMD RO_MSGFILTER
+#define FILTER_ARG msgfilter
+#define FILTER_ARG_SZ sizeof(msgfilter)
+#endif
+#if defined(ROUTE_FILTER) || defined(RO_MSGFILTER)
+        status = setsockopt(sNetlinkFd, AF_ROUTE, FILTER_CMD, FILTER_ARG, FILTER_ARG_SZ);
+        VerifyOrDie(status == 0, OT_EXIT_ERROR_ERRNO);
+#endif
+        status = fcntl(sNetlinkFd, F_SETFL, O_NONBLOCK);
+        VerifyOrDie(status == 0, OT_EXIT_ERROR_ERRNO);
+    }
+#endif // defined(__APPLE__) || defined(__NetBSD__) || defined(__FreeBSD__)
+}
+
+void platformNetifInit(otInstance *aInstance, const char *aInterfaceName)
+{
+    sIpFd = SocketWithCloseExec(AF_INET6, SOCK_DGRAM, IPPROTO_IP, kSocketNonBlock);
+    VerifyOrDie(sIpFd >= 0, OT_EXIT_ERROR_ERRNO);
+
+    otIcmp6SetEchoMode(aInstance, OT_ICMP6_ECHO_HANDLER_DISABLED);
+
```

Comment:
Do we need so many new lines here?

Reply:
I've tried to get rid of some of them, but I like code that is logically performing a set of related steps to be separated from code that is performing a different set of steps.

Reply:
I think this is better now.